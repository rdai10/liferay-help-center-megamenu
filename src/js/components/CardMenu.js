import preact from 'preact';
import PropTypes from 'prop-types';

import Card from './Card';

const CardMenu = ({className, configs, type}) => (
	<div class={className}>
		{configs.map(
			(config, index) => (
				<Card
					description={config.description}
					key={index}
					name={config.name}
					svgId={config.svgId}
					type={type}
					url={config.url}
				/>
			)
		)}
	</div>
);

CardMenu.defaultProps = {
	type: 'product'
};

CardMenu.propTypes = {
	className: PropTypes.string,
	configs: PropTypes.arrayOf(PropTypes.object),
	type: PropTypes.oneOf(['nav', 'product'])
};

export default CardMenu;