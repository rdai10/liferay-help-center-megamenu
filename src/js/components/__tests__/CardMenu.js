import preact from 'preact';
import {cleanup, render} from 'preact-testing-library';

import CardMenu from '../CardMenu';

afterEach(cleanup);

describe('CardMenu', () => {
	const configs = [
		{
			description: 'Description for One',
			name: 'One',
			svgId: '#search',
			url: '/'
		},
		{
			description: 'Description for Two',
			svgId: '#ticket',
			url: '/'
		},
		{
			name: 'Three',
			svgId: '#book',
			url: '/'
		},
		{
			description: 'Description for Four',
			name: 'Four',
			url: '/'
		},
		{
			description: 'Description for Five',
			name: 'Five',
			svgId: '#bug'
		}
	];

	it('renders product card menu correctly', () => {
		const {container} = render(<CardMenu configs={configs} />);

		expect(container).toMatchSnapshot();
	});

	it('renders nav card menu correctly', () => {
		const {container} = render(<CardMenu configs={configs} type="nav" />);

		expect(container).toMatchSnapshot();
	});
});
