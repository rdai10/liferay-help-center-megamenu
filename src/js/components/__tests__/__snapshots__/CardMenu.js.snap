// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMenu renders nav card menu correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="nav-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#search"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            One
          </h4>
          <p
            class="description"
          >
            Description for One
          </p>
        </div>
      </a>
    </div>
    <div
      class="nav-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#ticket"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          
          <p
            class="description"
          >
            Description for Two
          </p>
        </div>
      </a>
    </div>
    <div
      class="nav-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#book"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            Three
          </h4>
          
        </div>
      </a>
    </div>
    <div
      class="nav-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            Four
          </h4>
          <p
            class="description"
          >
            Description for Four
          </p>
        </div>
      </a>
    </div>
    <div
      class="nav-card"
    >
      <a
        class="autofit-row autofit-row-center"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#bug"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            Five
          </h4>
          <p
            class="description"
          >
            Description for Five
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`CardMenu renders product card menu correctly 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="product-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#search"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            One
            <svg
              class="lexicon-icon lexicon-icon-arrow-right"
            >
              <use
                href="#arrow-right"
              />
            </svg>
          </h4>
          <p
            class="description"
          >
            Description for One
          </p>
        </div>
      </a>
    </div>
    <div
      class="product-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#ticket"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          
          <p
            class="description"
          >
            Description for Two
          </p>
        </div>
      </a>
    </div>
    <div
      class="product-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#book"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            Three
            <svg
              class="lexicon-icon lexicon-icon-arrow-right"
            >
              <use
                href="#arrow-right"
              />
            </svg>
          </h4>
          
        </div>
      </a>
    </div>
    <div
      class="product-card"
    >
      <a
        class="autofit-row autofit-row-center"
        href="/"
      >
        
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            Four
            <svg
              class="lexicon-icon lexicon-icon-arrow-right"
            >
              <use
                href="#arrow-right"
              />
            </svg>
          </h4>
          <p
            class="description"
          >
            Description for Four
          </p>
        </div>
      </a>
    </div>
    <div
      class="product-card"
    >
      <a
        class="autofit-row autofit-row-center"
      >
        <div
          class="autofit-col"
        >
          <svg
            class="icon"
          >
            <use
              href="#bug"
            />
          </svg>
        </div>
        <div
          class="autofit-col autofit-col-expand"
        >
          <h4
            class="title"
          >
            Five
            <svg
              class="lexicon-icon lexicon-icon-arrow-right"
            >
              <use
                href="#arrow-right"
              />
            </svg>
          </h4>
          <p
            class="description"
          >
            Description for Five
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
